google/cloud/storage/bucket.py

```diff
@@ -1787,11 +1796,20 @@ def delete_blobs(
         for blob in blobs:
             try:
                 blob_name = blob
+                generation = None
                 if not isinstance(blob_name, str):
                     blob_name = blob.name
+                    generation = blob.generation if preserve_generation else None
+                else:
+                    if preserve_generation:
```

Comment:
In my opinion, we can do without this ValueError, since it's valid to pass a mixed list of blobs and strings and this will trigger if even one of the list is a string. Let me know if you disagree and we can discuss.

Reply:
Ooooh I've overlooked the possibility of passing a mixed list of blobs and strings. Considering this use case, it would make sense to remove the ValueError. Also like your proposed wording above, along with that description, I think that's clear!